### PR TITLE
Stop consensus on error

### LINKED ISSUE #864: Handle errors in `on_ready` in `consensus.rs`
We shouldn't let consensus `advance` to the new state if there were errors in handling previous changes.

### PR DESCRIPTION
**Pull request by e-ivkov**
Closes #864 

The failures can happen only in very rare corner cases, so I am bit unsure as to how to test it.

### CODE REVIEW

**Comment by agourlay** [comment]
It looks legit but TBH I am starting to have problems fitting the state machine in my head :sweat_smile:

**Comment by e-ivkov** [comment]
> It looks legit but TBH I am starting to have problems fitting the state machine in my head sweat_smile

More robustness - more complexity :man_shrugging: . That's the easiest solution that I came up with. Of course it would be better if the handled it as part of the raft lib. Maybe we can write a light wrapper around raft lib and move it into a separate crate for it to become simpler.

**Comment by generall** [comment]
I also feel like we need at least some document with explanation

File: lib/storage/src/content_manager/consensus_state.rs [hunk 1]
```diff
@@ -77,6 +81,13 @@ impl<C: CollectionContainer> ConsensusState<C> {
             on_consensus_op_apply: Default::default(),
             propose_sender,
             first_voter: Default::default(),
+            consensus_thread_status: RwLock::new(ConsensusThreadStatus::Working),
```
Comment (agourlay) [inline lib/storage/src/content_manager/consensus_state.rs:84]: So right now there are no places changing `consensus_thread_status` from error to working? 
Just want to make sure I get this :)

Comment (e-ivkov) [reply lib/storage/src/content_manager/consensus_state.rs:84]: Yes, if it fails - it stays in failed state until restart.

Comment (e-ivkov) [reply lib/storage/src/content_manager/consensus_state.rs:84]: The idea is to allow search and update operations on existing collections regardless of consensus errors. Retrying errors doesn't seem to make sense as they are mostly environment related and therefore unrecoverable from app point of view

Comment (agourlay) [reply lib/storage/src/content_manager/consensus_state.rs:84]: Thanks, it makes sense :+1:

**Review by agourlay — APPROVED** [review]

**Review by generall — APPROVED** [review]
